lib/dal/src/func/associations.rs

```diff
@@ -341,47 +358,47 @@ impl FuncAssociations {
             .collect())
     }
 
-    // async fn compile_leaf_function_input_types(
-    //     ctx: &DalContext,
-    //     schema_variant_ids: &[SchemaVariantId],
-    //     inputs: &[LeafInputLocation],
-    // ) -> FuncViewResult<String> {
-    //     let mut ts_type = "type Input = {\n".to_string();
-    //
-    //     for input_location in inputs {
-    //         let input_property = format!(
-    //             "{}?: {} | null;\n",
-    //             input_location.arg_name(),
-    //             Self::get_per_variant_types_for_prop_path(
-    //                 ctx,
-    //                 schema_variant_ids,
-    //                 &input_location.prop_path(),
-    //             )
-    //             .await?
-    //         );
-    //         ts_type.push_str(&input_property);
-    //     }
-    //     ts_type.push_str("};");
-    //
-    //     Ok(ts_type)
-    // }
-    //
-    // async fn get_per_variant_types_for_prop_path(
-    //     ctx: &DalContext,
-    //     variant_ids: &[SchemaVariantId],
-    //     path: &PropPath,
-    // ) -> FuncViewResult<String> {
-    //     let mut per_variant_types = vec![];
-    //
-    //     for variant_id in variant_ids {
-    //         let prop = Prop::find_prop_by_path(ctx, *variant_id, path).await?;
-    //         let ts_type = prop.ts_type(ctx).await?;
-    //
-    //         if !per_variant_types.contains(&ts_type) {
-    //             per_variant_types.push(ts_type);
-    //         }
-    //     }
-    //
-    //     Ok(per_variant_types.join(" | "))
-    // }
+    async fn compile_leaf_function_input_types(
```

Comment:
Let's compile some leaf input types. We should be able to do something similar for all attribute funcs, especially now that `Props` have the `ts_types` method.